ui/app/ducks/gas/gas.duck.js

```diff
@@ -161,6 +171,40 @@ async function fetchExternalBasicGasEstimates(dispatch) {
     setStorageItem('BASIC_PRICE_ESTIMATES_LAST_RETRIEVED', timeRetrieved),
   ]);
   dispatch(setBasicPriceEstimatesLastRetrieved(timeRetrieved));
+  return basicEstimates;
+}
+
+async function fetchEthGasPriceEstimates(state) {
+  const chainId = getCurrentChainId(state);
+  let timeLastRetrieved, cachedBasicEstimates;
+  await Promise.all([
+    getStorageItem(`${chainId}_BASIC_PRICE_ESTIMATES_LAST_RETRIEVED`),
+    getStorageItem(`${chainId}_BASIC_PRICE_ESTIMATES`),
+  ]).then((values) => {
+    timeLastRetrieved = values[0] ? values[0] : 0;
+    cachedBasicEstimates = values[1];
+  });
```

Comment:
Nit: It's typically better to avoid using `.then` altogether, and stick to just `await`, because `await` enables async stack traces on some platforms (e.g. Node.js and Chrome; not Firefox yet but they're working on it).

```suggestion
  let [cachedTimeLastRetrieved, cachedBasicEstimates] = await Promise.all([
    getStorageItem(`${chainId}_BASIC_PRICE_ESTIMATES_LAST_RETRIEVED`),
    getStorageItem(`${chainId}_BASIC_PRICE_ESTIMATES`),
  ]);
  const timeLastRetrieved = cachedTimeLastRetrieved ? cachedTimeLastRetrieved || 0;
```